src/diagnostics.rs

```diff
@@ -30,14 +30,18 @@ pub fn publish_diagnostics(params: Params, ctx: &mut Context) {
                 lsp_range_to_kakoune(&x.range, &document.text, ctx.offset_encoding),
                 match x.severity {
                     Some(DiagnosticSeverity::Error) => "DiagnosticError",
+                    Some(DiagnosticSeverity::Hint) => "DiagnosticHint",
+                    Some(DiagnosticSeverity::Information) => "DiagnosticInfo",
                     _ => "DiagnosticWarning",
```

Comment:
Since we're now implementing all possible values, the last arm should be explicit as well. That way, we'll get a compiler error if there ever was a change to this in an update of `lsp_types`.
```suggestion
                    Some(DiagnosticSeverity::Warning) => "DiagnosticWarning",
```

Reply:
I had kept it because one of the possible patterns is `None`, but you're right that it would just happily classify new diagnostic levels as warnings. I'll add an explicit case to convert `None` into warnings as well.

Reply:
Oh, right. I overlooked the fact that it's an `Option<DiagnosticSeverity>`.